FirebaseABTesting/Sources/ABTExperimentPayload.m

```diff
@@ -0,0 +1,122 @@
+static NSString *const kExperimentPayloadKeyTriggerTimeoutMillis = @"triggerTimeoutMillis";
+static NSString *const kExperimentPayloadKeyTimeToLiveMillis = @"timeToLiveMillis";
+static NSString *const kExperimentPayloadKeySetEventToLog = @"setEventToLog";
+static NSString *const kExperimentPayloadKeyActivateEventToLog = @"activateEventToLog";
+static NSString *const kExperimentPayloadKeyClearEventToLog = @"clearEventToLog";
+static NSString *const kExperimentPayloadKeyTimeoutEventToLog = @"timeoutEventToLog";
+static NSString *const kExperimentPayloadKeyTTLExpiryEventToLog = @"ttlExpiryEventToLog";
+static NSString *const kExperimentPayloadKeyOverflowPolicy = @"overflowPolicy";
+static NSString *const kExperimentPayloadKeyOngoingExperiments = @"ongoingExperiments";
+
+@implementation ABTExperimentLite
+
+- (instancetype)initWithExperimentId:(NSString *)experimentId {
+  if (self = [super init]) {
+    _experimentId = experimentId;
+  }
+  return self;
+}
+
+@end
+
+@implementation ABTExperimentPayload
+
++ (NSDateFormatter *)experimentStartTimeFormatter {
+  NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+  [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"];
+  [dateFormatter setTimeZone:[NSTimeZone timeZoneForSecondsFromGMT:0]];
+  // Locale needs to be hardcoded. See
+  // https://developer.apple.com/library/ios/#qa/qa1480/_index.html for more details.
+  [dateFormatter setLocale:[[NSLocale alloc] initWithLocaleIdentifier:@"en_US_POSIX"]];
+  [dateFormatter setTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"UTC"]];
+  return dateFormatter;
+}
+
++ (instancetype)parseFromData:(NSData *)data {
+  NSError *error;
+  NSDictionary *experimentDictionary =
+      [NSJSONSerialization JSONObjectWithData:data
+                                      options:NSJSONReadingAllowFragments
+                                        error:&error];
+  if (error != nil) {
+    return nil;
+  } else {
+    return [[ABTExperimentPayload alloc] initWithDictionary:experimentDictionary];
+  }
+}
+
+- (instancetype)initWithDictionary:(NSDictionary<NSString *, id> *)dictionary {
+  if (self = [super init]) {
+    // Strings.
+    _experimentId = dictionary[kExperimentPayloadKeyExperimentID];
+    _variantId = dictionary[kExperimentPayloadKeyVariantID];
+    _triggerEvent = dictionary[kExperimentPayloadKeyTriggerEvent];
+    _setEventToLog = dictionary[kExperimentPayloadKeySetEventToLog];
+    _activateEventToLog = dictionary[kExperimentPayloadKeyActivateEventToLog];
+    _clearEventToLog = dictionary[kExperimentPayloadKeyClearEventToLog];
+    _timeoutEventToLog = dictionary[kExperimentPayloadKeyTimeoutEventToLog];
+    _ttlExpiryEventToLog = dictionary[kExperimentPayloadKeyTTLExpiryEventToLog];
+
+    // Other.
+    NSDate *experimentStartTime = [[[self class] experimentStartTimeFormatter]
+        dateFromString:dictionary[kExperimentPayloadKeyExperimentStartTime]];
```

Comment:
nit, just want to make sure if dictionary[kExperimentPayloadKeyExperimentStartTime] is nil, it won't crash here.

Reply:
The changes in c2b6de3 prevent this.